spec/rspec/core/rake_task_spec.rb

```diff
@@ -36,14 +36,21 @@ def spec_command
 
     context "default" do
       it "renders rspec" do
-        expect(spec_command).to match(/^#{ruby} #{default_load_path_opts} #{task.rspec_path}/)
+        expect(spec_command).to match(/^#{ruby} #{default_load_path_opts} '?#{task.rspec_path}'?/)
+      end
+    end
+
+    context "with space", :unless => RSpec::Support::OS.windows? do
+      it "renders rspec with space escaped" do
+        task.instance_variable_set(:@rspec_path, '/path with space/exe/rspec')
```

Comment:
Hi,
On Sat, Jan 17, 2015 at 12:06:41AM -0800, Myron Marston wrote:
> > @@ -36,14 +36,21 @@ def spec_command
> >  
> >      context "default" do
> >        it "renders rspec" do
> > -        expect(spec_command).to match(/^#{ruby} #{default_load_path_opts} #{task.rspec_path}/)
> > +        expect(spec_command).to match(/^#{ruby} #{default_load_path_opts} '?#{task.rspec_path}'?/)
> > +      end
> > +    end
> > +
> > +    context "with space", :unless => RSpec::Support::OS.windows? do
> > +      it "renders rspec with space escaped" do
> > +        task.instance_variable_set(:@rspec_path, '/path with space/exe/rspec')
> 
> `rspec_path` is declared as an `attr_accessor`:
> 
> https://github.com/agx/rspec-core/blob/6650b33349b50b4d85dec63df25c89438ed958b7/lib/rspec/core/rake_task.rb#L46
> 
> Which means you should be able to assign it using a writer method
> rather than monkeying with internal instance variables:

> 
> ``` ruby
> task.rspec_path = "..."
> ```

That worked now - no idea what I didn't work when I tried it
yesterday. Sorry for the noise on such a small issue.